read-fonts/src/font_data.rs

```diff
@@ -24,6 +24,7 @@ pub struct FontData<'a> {
 /// # Note
 ///
 /// call `finish` when you're done to ensure you're in bounds
+#[derive(Debug, Default, Clone, Copy)]
```

Comment:
so... this is not an issue with this patch, but more an observation: something that has been bugging me about `Cursor`, so far, is that it is serving two purposes: it was initially designed specifically for the validation case, and then later on I started using it in places for actually constructing items.

These do feel like two distinct tasks, sort of? The validating cursor does need to be able to read scalars, but it also has a bunch of API for skipping items etc that might not be useful for a more general purpose cursor. Does any of this resonate with you?

Reply:
I didn't find it particularly cumbersome to use for this purpose and I did actually need to use the `read_array` method for grabbing the composite bytecode instructions, so I was happy that it was there. It's probably worth revisiting these types after we have a greater portion of the spec implemented and a better understanding of what is necessary.

I'll keep this in mind as we add new tables.